FnCAS/fncas/base.h

```diff
@@ -64,7 +64,23 @@ enum class type_t : uint8_t { variable, value, operation, function };
 enum class operation_t : uint8_t { add, subtract, multiply, divide, end };
 enum class function_t : uint8_t { sqr, sqrt, exp, log, sin, cos, tan, asin, acos, atan, unit_step, ramp, end };
 
+enum class JIT {
+  Super,          // The superclass of all `function_t<>`-s and `gradient_t<>`-s respectively.
+  Blueprint,      // The blueprint of the function of gradient, evaluated by traversing an internal tree.
+  NativeWrapper,  // A wrapper over an `std::function<double(std::vector<double>&)>`, or approximate gradient computer.
```

Comment:
Indeed they are not. I was hoping you'd suggest a better name for this `enum class`. :-)